src/VisualStudio/Core/Def/Implementation/TaskList/ProjectExternalErrorReporter.cs

```diff
@@ -284,19 +290,47 @@ public int ClearErrors()
             int originalStartLine,
             int originalStartColumn,
             int originalEndLine,
-            int originalEndColumn)
+            int originalEndColumn,
+            DiagnosticAnalyzerInfoCache analyzerInfoCache)
         {
+            string title, description, category, helpLink;
+            DiagnosticSeverity defaultSeverity;
+            bool isEnabledByDefault;
+            ImmutableArray<string> customTags;
+
+            if (analyzerInfoCache != null && analyzerInfoCache.TryGetDescriptorForDiagnosticId(errorId, out var descriptor))
+            {
+                title = descriptor.Title.ToString(CultureInfo.CurrentCulture);
+                description = descriptor.Description.ToString(CultureInfo.CurrentCulture);
+                category = descriptor.Category;
+                defaultSeverity = descriptor.DefaultSeverity;
+                isEnabledByDefault = descriptor.IsEnabledByDefault;
+                customTags = descriptor.CustomTags.AsImmutableOrEmpty();
+                helpLink = descriptor.HelpLinkUri;
+            }
+            else
+            {
+                title = message;
+                description = message;
+                category = WellKnownDiagnosticTags.Build;
+                defaultSeverity = severity;
+                isEnabledByDefault = true;
+                customTags = IsCompilerDiagnostic(errorId) ? CompilerDiagnosticCustomTags : CustomTags;
```

Comment:
`CompilerDiagnosticCustomTags` is set to what the compiler uses for warnings. For errors, the custom tags of a compiler diagnostic will have an extra "NotConfigurable" tag.

I have no idea if this affects anything.

Actually when can we hit this "else" block?